### ISSUE TITLE
new JDatabaseIteratorSqlsrv(*string*, cursor) 

### ISSUE THREAD

**Issue by rudisten1** [joomla/joomla-cms#11297]
#### Steps to reproduce the issue

install joomla using a SQL Server db as the main database
create a user account giving it the access level Registered
login in the front end with the newly created user account
#### Expected result

going to the default home page (in this case the featured articles)
#### Actual result

site error as follow:

An error has occurred while processing your request.

You may not be able to visit this page because of:

```
an out-of-date bookmark/favourite
a mistyped address
a search engine that has an out-of-date listing for this site
you have no access to this page
```

Go to the Home Page

Home Page

If difficulties persist, please contact the System Administrator of this site and report the error below.

```
0 new JDatabaseIteratorSqlsrv(*string*, cursor) 
```
#### System information (as much as possible)

php 7
 sqlsrv extension 4.0 non thread safe
 IIS 7
 joomla 3.6
 SQL Server 2014

**Comment by alikon** [comment]
are you using some 3party extensions?

**Comment by rudisten1** [comment]
no. I was able to find out that the issue arises only with php 7, not with php 5.6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11297">issues.joomla.org/joomla-cms/11297</a>.</sub>

**Comment by williamsandy** [comment]
Can confirm this happens with fresh install of Joomla 3.6.

I found that this issue is occurring because the JDatabaseIterator class constructor cannot find the stdClass in PHP 7.  The issue seems to resolve if you explicitly specify to PHP to look in the global namespace. Why you have to do this in this case? I don't know. I can only assumme some namespace is being used in one of the classes specific to SQL Server

So change in the JDatabaseDriver at libraries\joomla\database\driver.php 

on line 1242

public function getIterator($column = null, $class = 'stdClass')

to

public function getIterator($column = null, $class = '\stdClass')

This is enough to fix this specific issue . However you should update the JDatabaseIterator class as well to avoid a similar issue.

So change in JDatabaseDriver at libraries\joomla\database\iterator.php 

on line 76

public function __construct($cursor, $column = null, $class = 'stdClass')

to 

public function __construct($cursor, $column = null, $class = '\stdClass')<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11297">issues.joomla.org/joomla-cms/11297</a>.</sub>

**Comment by mbabker** [comment]
That shouldn't be necessary.  All code in the libraries/Joomla directory is
in the global namespace.  Seems weird that it's acting otherwise.

**Comment by williamsandy** [comment]
Did a quick test

It seems this works 

 $a = new stdClass;


This does not work

$a = new 'stdClass';

It only occurs when you supply the class name as string!
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11297">issues.joomla.org/joomla-cms/11297</a>.</sub>

**Comment by sweco-semhul** [comment]
Seeing the same problem on Joomla 3.6.4, PHP 7.0.12, Windows Server 2012 R2, IIS 8.5 MS SQL Server 2008 R2.
@williamsandy solution above solved the problem

**Comment by brianteeman** [comment]
Closed as we have a PR for testing #13082
